android/expoview/src/main/java/host/exp/exponent/notifications/NotificationHelper.java

```diff
@@ -369,7 +378,26 @@ public static void showNotification(
       final ExponentManifest exponentManifest,
       final Listener listener) {
     final ExponentNotificationManager manager = new ExponentNotificationManager(context);
-    final String experienceId = (String) details.get("experienceId");
+
+    // The experienceId field sent in push notification payloads corresponds to the id field
+    // in the manifest. This is used for scoping if scopeKey isn't present.
+    final String legacyExperienceId = (String) details.get("experienceId");
+
+    // The projectId must be either the scopeKey field or null! We use it to find the record
+    // matching the experience in ExponentDB. In ExponentDB the id field is determined by
+    // ExponentManifest.getExperienceId(manifest) which will use scopeKey if available or fall
+    // back to using the id field.
+    //
+    // ***WARNING**
+    //
+    // If scopeKey is not included in the manifest but is included in notification payloads
+    // then this may break notifications and crash the app when a notification is received.
```

Comment:
If we send scopeKey in the notification payloads and it is consistent with the manifest ID (still using `@user/slug`), shouldn't we be fine here?